src/components/Modal/Modal.spec.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+
+import Modal from './Modal';
+
+describe('<Modal />', () => {
+  const mockFunc = jest.fn();
+  const children = <div>Children</div>;
+  let wrapper;
+
+  let props = {
+    show: true,
+    hideModal: mockFunc,
+  };
+
+  beforeEach(() => {
+    wrapper = shallow(<Modal {...props}>{children}</Modal>);
+  });
+
+  it('should match the snapshot', () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should set body class "remove-scroll" and be on focus mode when prop show is true', () => {
+    wrapper = mount(<Modal {...props}>{children}</Modal>);
+    expect(document.body.classList.contains('remove-scroll')).toBeTruthy();
+    expect(wrapper.is(':focus')).toBeTruthy();
+  });
+
+  it('should call mockFunc on click', () => {
+    const mockEvent = {
+      target: {
+        classList: {
+          contains: () => 'modal'
+        },
+      },
+    };
+
+    wrapper.simulate('click', mockEvent);
+    expect(mockFunc).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call mockFunc on escape button', () => {
+    const mockEvent = {
+      target: {
+        keyCode: 27
+      },
+    };
+
+    wrapper.simulate('keyDown', mockEvent);
+    expect(mockFunc).toHaveBeenCalledTimes(1);
+  });
+
+  it('should not set body class "remove-scroll" when prop show is false', () => {
+    props.show = false;
+    wrapper = mount(<Modal {...props}>{children}</Modal>);
+    expect(document.body.classList.contains('remove-scroll')).toBeFalsy();
+  });
+});
```

Comment:
Very nice!

Reply:
I'm not testing expert, but i guess, this is not the best approach to test your component. If you change classname behavior will be the same, but tests will fail, which means you're testing not behavior(which is more preferable) but the implementation, and it blocks your flexibility in refactoring.